src/api/v2/v2AzureResourcesApi.ts

```diff
@@ -82,6 +82,21 @@ export interface WrappedResourceModel {
  * The interface that resource resolvers must implement
  */
 export interface BranchDataProvider<TResource extends ResourceBase, TModel extends ResourceModelBase> extends vscode.TreeDataProvider<TModel> {
+    /**
+     * @deprecated This is implemented by the Resources API and does not need to be implemented by the
+     * {@link BranchDataProvider}.
+     */
+    getParent?: never;
```

Comment:
In line with my suggestion, we can override with this `getParent?: never` to indicate that `BranchDataProvider`s do not need to implement `getParent` (and, _should_ not, to avoid confusion).